agents/processor/src/processor.rs

```diff
@@ -344,12 +341,12 @@ impl NomadAgent for Processor {
         Self: Sized,
     {
         Ok(Self::new(
-            settings.interval.parse().expect("invalid integer"),
+            settings.agent.interval,
             settings.as_ref().try_into_core(AGENT_NAME).await?,
-            settings.allowed,
-            settings.denied,
-            settings.indexon.is_some(),
-            settings.s3,
+            settings.agent.allowed,
+            settings.agent.denied,
+            settings.agent.index_only,
```

Comment:
did the semantics of this field change?

Reply:
Yes in the config crate `interval` is a `u64` and not string so skip parsing. `index_only` became `bool` rather than relying on `Option<String>` as semantic so skip `.is_some()` check.